http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/NodepoolComponent.scala

```diff
@@ -165,6 +172,17 @@ object nodepoolQuery extends TableQuery(new NodepoolTable(_)) {
       nodepools <- findByNodepoolIdQuery(id).result
     } yield nodepools.map(rec => unmarshalNodepool(rec, List())).headOption
 
+  def getMinimalByUser(creator: WorkbenchEmail,
+                       project: GoogleProject)(implicit ec: ExecutionContext): DBIO[Option[Nodepool]] =
+    for {
+      cluster <- kubernetesClusterQuery.getMinimalActiveClusterByName(project)
+    } yield cluster.flatMap(
+      _.nodepools
```

Comment:
Minor - but I think this selects all nodepools for the cluster from the DB, and then does the filtering in Scala. Could we do the filtering in the DB query instead?

Maybe something like:
```
    joinMinimalClusterAndUnmarshal(
      findActiveByNameQuery(project),
      nodepoolQuery.filter(_.destroyedDate === dummyDate).filterNot(_.isDefault).filter(_.creator === creator)
    ).map(
      _.flatMap(_.nodepools).headOption
    )
```